src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/client/ModuleBaritone.kt

```diff
@@ -0,0 +1,352 @@
+                        createSetting("Safe", assumeSafeWalk)
+                    }
+                }
+            }
+        }
+    }
+
+    private object Mining : Configurable("Mining") {
+        init {
+            if (BaritoneUtil.isAvailable) {
+                with(BaritoneAPI.getSettings()) {
+                    createSetting("MinYLevel", minYLevelWhileMining, maxRangedValue = 2031)
+                    createSetting("MaxYLevel", maxYLevelWhileMining, maxRangedValue = 2031)
+                    createSetting("GoalUpdateInterval", mineGoalUpdateInterval, maxRangedValue = 20)
+                    createSetting("MaxOreLocationsCount", mineMaxOreLocationsCount, maxRangedValue = 64)
+                    createSetting("PauseForFallingBlocks", pauseMiningForFallingBlocks)
+                    createSetting("ForceInternal", forceInternalMining)
+                    createSetting("OnlyExposedOres", allowOnlyExposedOres)
+                    createSetting("UseSword", useSwordToMine)
+                    createSetting("NotificationOnFail", notificationOnMineFail)
+                }
+
+                treeAll(
+                    Legit
+                )
+            }
+        }
+
+        private object Legit : Configurable("Legit") {
+            init {
+                if (BaritoneUtil.isAvailable) {
+                    with(BaritoneAPI.getSettings()) {
+                        createSetting("Allow", legitMine)
+                        createSetting("IncludeDiagonals", legitMineIncludeDiagonals)
+                        createSetting("YLevel", legitMineYLevel, -64, 320)
+                    }
+                }
+            }
+        }
+    }
+
+    private object Movement : Configurable("Movement") {
+        init {
+            if (BaritoneUtil.isAvailable) {
+                with(BaritoneAPI.getSettings()) {
+                    createSetting("Sprint", allowSprint)
+                    createSetting("JumpAt256", allowJumpAt256)
+                }
+
+                treeAll(
+                    Ascends,
+                    Descends,
+                    Parkour
+                )
+            }
+        }
+
+        private object Ascends : Configurable("Ascends") {
+            init {
+                if (BaritoneUtil.isAvailable) {
+                    with(BaritoneAPI.getSettings()) {
+                        createSetting("Diagonal", allowDiagonalAscend)
+                        createSetting("WithSprint", sprintAscends)
+                        createSetting("Parkour", allowParkourAscend)
+                    }
+                }
+            }
+        }
+
+        private object Descends : Configurable("Descends") {
+            init {
+                if (BaritoneUtil.isAvailable) {
+                    with(BaritoneAPI.getSettings()) {
+                        createSetting("Diagonal", allowDiagonalDescend)
+                        createSetting("OvershootDiagonal", allowOvershootDiagonalDescend)
```

Comment:
Not sure how this would work with your system, but Diagonal should be a ToggleableConfigurable with OvershootDiagonal renamed to Overshoot as an setting there

Reply:
It needs a little rewriting, but it'll take time, I'll see if it's worth it.